Added configuration options

MatthewTamlin opened the pull request:
Added configuration options to allow scripts to configure the following properties in Android projects:
- Enable/disable HTML report generation
- Enable/disable HTML report copying to assets
- Enable/disable JSON report generation
- Enable/disable JSON report copying to assets

Added tests to check that options are used by task.

jaredsburrows commented on src/main/groovy/com/jaredsburrows/license/LicensePlugin.groovy:
```diff
@@ -30,6 +30,8 @@ final class LicensePlugin implements Plugin<Project> {
     // Get correct plugin - Check for android library, default to application variant for application/test plugin
     final variants = getAndroidVariants(project)
 
+    final configurationExtension = project.extensions.create("options", AndroidLicenseReportOptions)
```
We should probably change `options` to something more clear such as `licenseReport` or just `license`.

jaredsburrows commented on src/test/groovy/com/jaredsburrows/license/LicenseReportTaskAndroidSpec.groovy:
```diff
@@ -635,4 +635,102 @@ final class LicenseReportTaskAndroidSpec extends BaseAndroidSpecification {
+      compile DESIGN
+    }
+
+    project.extensions.configure(AndroidLicenseReportOptions, {options -> options.generateHtmlReport = false})
```
Can you use the DSL directly?

```
project.extensions.options {
  generateHtmlReport = false
  generateJsonReport = false
  copyHtmlReportToAssets = copyEnabled
  copyJsonReportToAssets = copyEnabled
}
```

jaredsburrows commented:
This is look great just a few comments and we need to update READMME.md for documentation.

MatthewTamlin commented:
I've made some changes to the PR.
- The tests now directly use the extension name.
- I've renamed the extension to 'licenseReport'. I didn't realise this was the name that was exposed for use in the build script.
- I've enabled configuration of Java projects in addition to Android projects.
- Javadoc updates.
- Added info to the README.

Let me know about the README placement. I wasn't sure where the best place was.

jaredsburrows commented:
Released `0.8.0` and so this will be in the `0.8.1-SNAPSHOT`.

jaredsburrows commented:
Thanks!
